### ISSUE TITLE
Custom python distribution links

### ISSUE THREAD

**Issue by kotborealis** [indygreg/PyOxidizer#589]
pyOxidizer downloads python binaries from github releases (see [default_python_distributions.rs](https://github.com/indygreg/PyOxidizer/blob/739b5f207388445955e37c28be6fc787c0c6e45b/pyoxidizer/src/default_python_distributions.rs])).

I want to selfhost those artifacts, is there any way to specify custom download links during pyoxidizer runtime?

**Comment by indygreg** [comment]
Yes, config files can specify the URL of a distribution to fetch by constructing a `PythonDistribution` instance from Starlark instead of calling `default_python_distribution()`. See https://gregoryszorc.com/docs/pyoxidizer/0.22.0/pyoxidizer_config_type_python_distribution.html.

A potential feature would be to change the root URL for the default Python distributions. Would that feature be of interest to you?

**Comment by kotborealis** [comment]
Thanks!

Yeah, such feature would be useful, for example, in air-gapped environments, where you can't download something from the internet during build.
